src/gui/socketapi.cpp

```diff
@@ -555,22 +565,36 @@ void SocketApi::command_GET_STRINGS(const QString &argument, SocketListener *lis
 void SocketApi::command_GET_MENU_ITEMS(const QString &argument, OCC::SocketListener *listener)
 {
     listener->sendMessage(QString("GET_MENU_ITEMS:BEGIN"));
+    auto flagString = [](bool enabled) { return enabled ? QLatin1String("::") : QLatin1String(":d:"); };
+
     bool hasSeveralFiles = argument.contains(QLatin1Char('\x1e')); // Record Separator
     Folder *syncFolder = hasSeveralFiles ? nullptr : FolderMan::instance()->folderForPath(argument);
-    if (syncFolder) {
+    if (syncFolder && syncFolder->accountState()->isConnected()) {
         QString systemPath = QDir::cleanPath(argument);
         if (systemPath.endsWith(QLatin1Char('/'))) {
             systemPath.truncate(systemPath.length() - 1);
         }
-        QString relativePath = systemPath.mid(syncFolder->cleanPath().length() + 1);
 
         SyncJournalFileRecord rec;
-        if (syncFolder->accountState()->isConnected() && syncFolder->journalDb()->getFileRecord(relativePath, &rec) && rec.isValid()) {
+        QString relativePath = systemPath.mid(syncFolder->cleanPath().length() + 1);
+        bool isOnTheServer = syncFolder->journalDb()->getFileRecord(relativePath, &rec) && rec.isValid();
+
+        listener->sendMessage(QLatin1String("MENU_ITEM:OPEN_PRIVATE_LINK") + flagString(isOnTheServer) + tr("Open in browser"));
```

Comment:
Minor: I think it would look nicer if this was the last option in the menu, because it belongs with the other "private link" actions.

Reply:
I put it first because I thought it would be the most commonly used action.  But I can put it last.